_data/import-public.rb

```diff
@@ -12,6 +12,23 @@
 require 'hash-joiner'
 require 'safe_yaml'
 
+PROJECT_FILES = Dir["private/projects/*.yml"]
+
+private_projects_yaml_file = "private/projects.yml"
+project_data = SafeYAML.load_file(private_projects_yaml_file, :safe=>true)
+PROJECT_FILES.each do |source|
+  data = SafeYAML.load_file(source, :safe=>true)
+  unless data
+    puts "Failed to parse #{source}"
+    exit 1
+  end
+  data = data.first  # one project per yaml file in projects dir
+  index = project_data.index { |proj| proj['name'] == data['name'] }
+  index = project_data.length  if index.nil?
+  project_data[index] = data
+end
+open(private_projects_yaml_file, 'w') {|outfile| outfile.puts project_data.to_yaml}
```

Comment:
Let me see if I understand what `HashJoiner` is doing here. 

Given `HashJoiner.join_array_data 'name', project_data, project_files_data` and data files:
```
projects.yml
---
- project: myRA
  name: myra
  github: 18F/myra
```
and:
```
projects/myra.yml
---
- project: New name for myRA
  name: new_myra
  github:
  - 18F/myra
  - 18F/other-repo
```
The new hash will be identical to the one in the projects directory?